### PR TITLE
Introduce EdDSA-api plugin

### PR DESCRIPTION
**Pull request by jtnord**
* Introduce the [EdDSA-api plugin](https://github.com/jenkinsci/eddsa-api-plugin)
* ~Do not pass the Fat War to the JTH to avoid random issues with plugins being present (yet missing classes) that should not be present, or present and unexpected (ie failing when a given system property is set by JTH)~

### Testing done

None, to be tested in CI

### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Review by alecharp — APPROVED** [review]
If the weekly-test is not reporting issues, this looks good.

**Comment by jtnord** [comment]
> If the weekly-test is not reporting issues, this looks good.

I expect issues similar to [this](https://ci.jenkins.io/job/Tools/job/bom/job/PR-3218/4/testReport/hudson.plugins.emailext/MailAccountFIPSTest/pct_email_ext_plugin_weekly___testFormValidationForInsecureAuth/) (for which an issue will be filed along with a fix)

**Comment by alecharp** [comment]
Just as you said, it fails exactly where you said it would.

**Comment by jtnord** [comment]
Happy to split https://github.com/jenkinsci/bom/pull/3230/commits/72bada75804d903f077d3167b4d46e40b5fd0c2b into a separate PR (or indeed for someone to want to run a full build on this).

The same fix has been run in a full mode in https://github.com/jenkinsci/bom/pull/3218 (which has eddsa-api plugin as well as incrementals of `mina-api` and `trilead-api`)

File: pct.sh [hunk 1]
```diff
@@ -28,7 +28,6 @@ exec java \
 	--working-dir "$(pwd)/target/pct-work" \
 	$PCT_D_ARGS \
 	${PCT_OPTS-} \
-	-Djth.jenkins-war.path="$(pwd)/target/megawar-$LINE.war" \
```
Comment (jglick) [inline pct.sh:31]: This is a far-reaching change that should not be done without discussion.

**Review by jglick — CHANGES_REQUESTED** [review]

File: pct.sh (hunk 1 above)
Comment (jtnord) [reply pct.sh:31]: it is broken by design :)

Comment (jtnord) [reply pct.sh:31]: https://github.com/jenkinsci/bom/issues/3231 should be used for any discussion about this.

Comment (jtnord) [reply pct.sh:31]: split removing the JTH to https://github.com/jenkinsci/bom/pull/3232

**Comment by jglick** [comment]
The particular error mentioned in #3231 could be solved in `email-ext` by switching `MailAccountFIPSTest` to use `JenkinsSessionRule` and then `Assume` that the megawar property is unset before continuing with the test (`then`).

File: pct.sh (hunk 1 above)
Comment (MarkEWaite) [reply pct.sh:31]: Merged in

* #3232

**Review by MarkEWaite — APPROVED** [review]

File: sample-plugin/pom.xml [hunk 2]
```diff
@@ -33,6 +33,10 @@
       <groupId>io.jenkins.plugins</groupId>
       <artifactId>byte-buddy-api</artifactId>
     </dependency>
+    <dependency>
+      <groupId>io.jenkins.plugins</groupId>
+      <artifactId>eddsa-api</artifactId>
```
Comment (jglick) [inline sample-plugin/pom.xml:38]: Note: the original intent of this file was to include only dependencies that were actually used “naturally”, not transitive deps. This was never mechanically enforced, though, and has proven difficult to manage: #683

**Review by jglick — APPROVED** [review]

**Pull request merged**